arcgis-ios-sdk-samples/Features/Generate geodatabase/GenerateGeodatabaseViewController.swift

```diff
@@ -59,16 +59,16 @@ class GenerateGeodatabaseViewController: UIViewController {
                     return
                 }
                 
-                for index in featureServiceInfo.layerInfos.indices.reversed() {
-                    let layerInfo = featureServiceInfo.layerInfos[index]
-                    //For each layer in the serice, add a layer to the map
-                    let layerURL = self.syncTask.url!.appendingPathComponent(String(index))
+                // For each layer in the service, add a layer to the map.
+                let featureLayers = featureServiceInfo.layerInfos.enumerated().map { (offset, layerInfo) -> AGSFeatureLayer in
+                    let layerURL = self.syncTask.url!.appendingPathComponent(String(offset))
                     let featureTable = AGSServiceFeatureTable(url: layerURL)
                     let featureLayer = AGSFeatureLayer(featureTable: featureTable)
                     featureLayer.name = layerInfo.name
                     featureLayer.opacity = 0.65
-                    self.mapView.map?.operationalLayers.add(featureLayer)
+                    return featureLayer
                 }
+                self.mapView.map?.operationalLayers.addObjects(from: featureLayers.reversed())
```

Comment:
Do you think this will take a hit on performance by making the intermediate `featureLayers`?

Reply:
I'm not certain, but methinks the benefits of updating `operationalLayers` with all the feature layers at once far outweighs the cost of the two intermediate arrays (`featureLayers` and the reversed array). Both are able to be allocated all at once because both `map(_:)` and `reversed()` know the size of the new array based on the old array.

Reply:
ok